examples/complex_roots.jl

```diff
@@ -0,0 +1,50 @@
+# Example of finding roots of a complex function
+# f(z) = z^4 + z - 2  (book of Hansen-Walster, chap. 10)
+
+using IntervalRootFinding, IntervalArithmetic
+
+"""
+    complex_bisection(f, X)
+
+Find complex roots of $f: \mathbb{C} \to \mathbb{C}$.
+
+Inputs:
+
+- `f`: function that takes $z \in \mathbb{C}$ and returns another
+complex number.
+
+- `X`: An `IntervalBox` specifying the bounds on the real and imaginary parts
+of `z`.
+
+"""
+function complex_bisection(f, X::IntervalBox)
+
+    """
+    Make a 2D real version of the complex function `f` suitable for `bisection`,
+    i.e. that accepts an `IntervalBox` and returns an `IntervalBox`
+    """
+    function g(X::IntervalBox)
+        x, y = X
+        z = x + im*y;
```

Comment:
Not really transcendental, but I read elsewhere that this is better written as `z = complex(x, y)` (avoids one multiplication and one addition).